includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -685,12 +687,44 @@ static function ( $validation_error ) {
 				 * @var DOMAttr $attr_node
 				 * @var string  $error_code
 				 */
-				list( $attr_node, $error_code ) = $disallowed_attribute;
-				$validation_error['code']       = $error_code;
+				list( $attr_node, $error_code, $error_data ) = $disallowed_attribute;
+				$validation_error['code']                    = $error_code;
 
-				$attr_spec = isset( $merged_attr_spec_list[ $attr_node->nodeName ] ) ? $merged_attr_spec_list[ $attr_node->nodeName ] : [];
-				if ( $this->remove_invalid_attribute( $node, $attr_node, $validation_error, $attr_spec ) ) {
-					$removed_attributes[] = $attr_node;
+				if ( self::DISALLOWED_PROPERTY_IN_ATTR_VALUE === $error_code ) {
+					$properties         = $error_data['original_properties'];
+					$invalid_properties = $error_data['invalid_properties'];
+
+					foreach ( $invalid_properties as $invalid_property ) {
+						$validation_error['attr_property_name'] = $invalid_property;
+						if ( $this->should_sanitize_validation_error( $validation_error, [ 'node' => $attr_node ] ) ) {
+							unset( $properties[ $invalid_property ] );
```

Comment:
While this complicates the logic here, removal of the invalid properties needs to be deferred later outside of this method. The reason is that if the removal is done too early, then if you opt to remove or keep the invalid property then this will impact the validation error objects for subsequent checks for invalid attributes. This can result in the latter validation errors needing to be re-removed after changeling the status of former validation errors.

You can see the logic later for this in relation to removal of invalid attributes. In particular, you can see it gathers up all the attributes and then passes them into the sanitize method to check if it should be removed. This ensures there is consistency across the validation errors as one is removed.

Alternatively perhaps the array of original attribute values could be obtained _before_ even this method is called, and then it wouldn't matter.

Reply:
@pierlon I'm sorry, I believe I am mistaken here. I first thought this diff was part of the `sanitize_disallowed_attribute_values_in_node` method but I see it is not. It looks like you are doing the right thing. I can see that `$validation_error['element_attributes']` has already been populated before this, so my concern was incorrect.